src/pull-retry.ts

```diff
@@ -0,0 +1,94 @@
+/*!
+ * Copyright 2019 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {StatusObject, status} from 'grpc';
```

Comment:
So I think we want to be using `@grpc/grpc-js` here

Reply:
I think we might want to stay on plain `grpc`? A need for channel pooling (via grpc-gcp) recently emerged and `grpc-js` is not supported yet, so we may be stuck on this version here for a little while still.

Reply:
For now that's totally reasonable.  You may need to monkey a little with the settings passed to gax or nodejs-common-grpc.  @alexander-fenster may be able to help us figure that out :)